### DISCUSSION TITLE
quasar build (pwa) with custom folder name?

### DISCUSSION THREAD

**Discussion by ssuess** [quasarframework/quasar#8966] [CLI - PWA mode] [answered]
Is there a way to customize the output folder name for a build command? For example, I would like to build beta and prod versions of my pwa and have them output to something like `/dist/prod` and `/dist/beta`. So some command like `quasar build -m pwa --folder prod` or whatever (could not find anything about this on https://quasar.dev/quasar-cli/commands-list)

**Comment by metalsadman** [comment]
you can use [CopyWebPackPlugin](https://webpack.js.org/plugins/copy-webpack-plugin/), then use `script` in your package json to pass a environment variable that you can then check at your `quasar.conf.js`
rough example: *did not test*
```js
// quasar.conf.js
extendWebpack (cfg) {
  ...
    if (ctx.prod) {
      cfg.plugins.push(
            new CopyWebpackPlugin({
              patterns: [
                {
                  from: path.resolve(__dirname, './dist'), // better to change your `distDir` as this gets cleaned up every time you run quasar build
                  to: './dist/' + process.env.SOME_VAR
                }
              ]
            })
          )
  }
```
ie. package.json>scripts>`"build-pwa-beta": "cross-env SOME_VAR=beta quasar build -m pwa"`
then run it `yarn build-pwa-beta`.

**Comment by rstoenescu** [comment, marked as answer]
quasar.conf.js > build > distDir (set this to a relative path to project root directory) https://quasar.dev/quasar-cli/quasar-conf-js#property-build

You can even configure it dynamically based on an console ENV variable:

```js
// $ BETA=true quasar build -m pwa (will output to dist/beta)
// $ quasar build -m pwa (will output to dist/prod)

// quasar.conf.js
build: {
  distDir: process.env.BETA ? 'dist/beta' : 'dist/prod'
```
